commons-packet/commons-packet-service/src/main/java/io/mosip/commons/packetmanager/service/PacketReaderService.java

```diff
@@ -1,261 +1,301 @@
+                    return getSource(jsonObject, process, key.toString());
+                }
+            }
+        }
+        return null;
+    }
+
+    private String getSource(JSONObject jsonObject, String process, String field) {
+        String source = null;
+        Object obj = field == null ? jsonObject.get(PROVIDER) : getField(jsonObject, field);
+        if (obj != null && obj instanceof ArrayList) {
+            List<String> providerList = (List) obj;
+            for (String value : providerList) {
+                String[] values = value.split(",");
+                for (String provider : values) {
+                    if (provider != null) {
+                        if (provider.startsWith(PROCESS) && provider.contains(process)) {
+                            for (String val : values) {
+                                if (val.startsWith(SOURCE)) {
+                                    return val.replace(SOURCE + ":", "").trim();
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return source;
+    }
+
+    private Object getField(JSONObject jsonObject, String field) {
+        LinkedHashMap lm = (LinkedHashMap) jsonObject.get(field);
+        return lm.get(PROVIDER);
+    }
+
+    private static JSONObject getJSONObject(JSONObject jsonObject, Object key) {
+        if(jsonObject == null)
+            return null;
+        LinkedHashMap identity = (LinkedHashMap) jsonObject.get(key);
+        return identity != null ? new JSONObject(identity) : null;
+    }
+
+    private JSONObject getMappingJsonFile() throws IOException {
+        if (mappingJson != null)
+            return mappingJson;
+
+        String mappingJsonString = restTemplate.getForObject(configServerUrl + "/" + mappingjsonFileName, String.class);
+        JSONObject jsonObject = objectMapper.readValue(mappingJsonString, JSONObject.class);
+        LinkedHashMap combinedMap = new LinkedHashMap();
+        combinedMap.putAll((Map) jsonObject.get(IDENTITY));
+        combinedMap.putAll((Map) jsonObject.get(DOCUMENTS));
+        combinedMap.put(META_INFO, jsonObject.get(META_INFO));
+        combinedMap.put(AUDITS, jsonObject.get(AUDITS));
+        mappingJson = new JSONObject(combinedMap);
+        return mappingJson;
+    }
+    public TagResponseDto getTags(TagRequestDto tagRequestDto) {
+    	try {
+			Map<String, String> tags = new HashMap<String, String>();
+			Map<String, String> existingTags = packetReader.getTags(tagRequestDto.getId());
+			List<String> tagNames=tagRequestDto.getTagNames();
+		    TagResponseDto tagResponseDto = new TagResponseDto();
+			if (tagNames != null && !tagNames.isEmpty()) {
+				for (String tag : tagNames) {
+					if (existingTags.containsKey(tag)) {
+						tags.put(tag, existingTags.get(tag));
+					} else {
+						throw new GetTagException(PacketUtilityErrorCodes.TAG_NOT_FOUND.getErrorCode(),

```

Comment:
As per the existing code base, if a tag is not found, do we throw an error ?

Reply:
Yes we are throwing the error